src/mono/mono/mini/mini.c

```diff
@@ -3178,8 +3178,8 @@ mini_method_compile (MonoMethod *method, guint32 opts, JitFlags flags, int parts
 	cfg->jit_mm = jit_mm_for_method (cfg->method);
 	cfg->mem_manager = m_method_get_mem_manager (cfg->method);
 
-	if (cfg->method->wrapper_type == MONO_WRAPPER_ALLOC) {
-		/* We can't have seq points inside gc critical regions */
+	if (cfg->method->wrapper_type == MONO_WRAPPER_ALLOC || cfg->method->wrapper_type == MONO_WRAPPER_NATIVE_TO_MANAGED) {
```

Comment:
What are these seq points, and how do the changes to the native_to_managed wrappers make it so that seq points cannot be in the wrapper? or is it something that should have been there originally?

Reply:
Sequence points needs a GOT entry and GOT entries are initialized when you load AOT module during runtime initialization, but since the native-to-managed wrapper can get called before runtime is initialized, it would crash if it includes sequence points before it gets to the call of the callback. The code that setup sequence points are done in function prolog, so no way to make the call to the callback before that. We never run soft debugger on these wrappers anyways, so they don't need to have sequence points enabled.